workitem/workitem_repository.go

```diff
@@ -1167,9 +1168,13 @@ func (r *GormWorkItemRepository) LoadByIteration(ctx context.Context, iterationI
 	return workitems, nil
 }
 
-// ChangeWorkItemType changes the workitem in wiStorage to newWIType. Returns
-// error if the operation fails
-func (r *GormWorkItemRepository) ChangeWorkItemType(ctx context.Context, wiStorage *WorkItemStorage, oldWIType *WorkItemType, newWIType *WorkItemType, spaceID uuid.UUID) error {
+// ChangeWorkItemType changes the workitem in wiStorage to newWIType.
+// If the updated workitem (from payload data) contains value for fields in
+// newWIType, that value will be set for the field. Field Values in payload
+// (updatedWorkitem) have precedence over the values in the old workitem (the
```

Comment:
The sentence with "precedence" should go away from comments and PR description. This is much more confusing than helping. As with a usual WI update, we try to set the fields according to the given type. That's all.

Reply:
Done in https://github.com/fabric8-services/fabric8-wit/pull/2254/commits/4a96083593dfe69c89a3a6669a7f4c2462c07bcd

I have kept some part of the comment because I want to make sure we don't have any confusion.